be/src/runtime/sender_queue.cpp

```diff
@@ -775,7 +770,7 @@ Status DataStreamRecvr::PipelineSenderQueue::add_chunks(const PTransmitChunkPara
             _chunk_queue_states[index].blocked_closure_num += closure != nullptr;
             _total_chunks++;
             // Double check here for short circuit compatibility without introducing a critical section
-            if (_chunk_queue_states[index].is_short_circuited.load(std::memory_order_relaxed)) {
+            if (_is_cancelled || _chunk_queue_states[index].is_short_circuited.load(std::memory_order_relaxed)) {
                 short_circuit(index);
```

Comment:
no need break?

Reply:
Yes, the logic is shared by short circuit and cancel. Short circuit may occur for some of parallelism, and the other parallelism need to proceed.